File: 9.PalindromeNumber.py
import math

class Solution:
    def isPalindrome(self, x: int) -> bool:
        if (x <0):
            return False
        rev, z = 0, x
        while (x) :
            rev = rev*10 + (x % 10);
            x = math.floor(x/10)
        return z == rev
    

s= Solution()
print(s.isPalindrome(-101))

File: 9.PalindromeNumber.js

var isPalindrome = function(x) {
    let reverseN = 0;
    if (x<0) return false
    const z = x
    while (x) {
        reverseN = reverseN*10 + (x % 10);
        x = Math.floor(x/10);
    }
    return z === reverseN;
};

console.log(isPalindrome(121))